Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,417 @@
+
+            for (int i = 0; i < scenePaths.Length; i++)
+            {
+                MigrateScene(scenePaths[i]);
+            }
+        }
+
+        private void MigrateScene(String path)
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+
+            var previousScenePath = EditorSceneManager.GetActiveScene().path;
+            Scene currentScene = EditorSceneManager.OpenScene(path);
+
+            foreach (var parent in currentScene.GetRootGameObjects())
+            {
+                MigrateGameObjectHierarchy(parent);
+            }
+            EditorSceneManager.SaveScene(currentScene);
+            EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
+        }
+
+        private void MigrateAllPrefabs()
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+
+            foreach (var path in FindAllAssets(PrefabExtension))
+            {
+                using (var editScope = new EditPrefabAsset(path))
+                {
+                    MigrateGameObjectHierarchy(editScope.root);
+                }
+            }
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+        }
+
+        private void MigrateSelection()
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+
+            foreach (var migrationObject in migrationObjects)
+            {
+                string path = AssetDatabase.GetAssetPath(migrationObject);
+
+                if (IsAsset(path))
+                {
+                    if (IsExtension(path, PrefabExtension))
+                    {
+                        // Modify and save Prefab Asset
+                        using (var editScope = new EditPrefabAsset(path))
+                        {
+                            GameObject parent = editScope.root;
+                            MigrateGameObjectHierarchy(parent);
+                        }
+                    }
+                    else if (IsExtension(path, SceneExtension))
+                    {
+                        MigrateScene(path);
+                    }
+                }
+                else
+                {
+                    MigrateGameObjectHierarchy((GameObject)migrationObject);
+                }
+            }
+            migrationObjects.Clear();
+        }
+
+        private void MigrateGameObjectHierarchy(GameObject gameObject)
+        {
+            foreach (var child in gameObject.GetComponentsInChildren<Transform>())
+            {
+                if (migrationTypeInstance.CanMigrate(child.gameObject))
+                {
+                    migrationTypeInstance.Migrate(child.gameObject);
+                }
+            }
+        }
+
+        private string[] FindAllAssets(string extension)
+        {
+
+            string[] scenePaths = Directory.GetFiles(Application.dataPath, "*" + extension, SearchOption.AllDirectories);
```

Comment:
Nit: filePaths (they're not necessarily scenePaths because extension can be arbitrary)